metal/eip_controlplane_reconciliation.go

```diff
@@ -106,9 +106,13 @@ func (m *controlPlaneEndpointManager) reconcileNodes(ctx context.Context, nodes
 		return err
 	}
 	resp, err := m.httpClient.Do(req)
+	// if there was no error, ensure we close
+	if err == nil && resp.Body != nil {
```

Comment:
sounds like the `resp.Body` check may not be needed here:
https://stackoverflow.com/a/33238755/156674
https://golang.org/pkg/net/http/#Client.Do
> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close.

Reply:
You are right, but it doesn't hurt.